src/EntityComponentManager_TEST.cc

```diff
@@ -2009,6 +2009,112 @@ TEST_P(EntityComponentManagerFixture, SetChanged)
       manager.ComponentState(e2, c2.first));
 }
 
+//////////////////////////////////////////////////
+TEST_P(EntityComponentManagerFixture, SerializedStateMapMsgAfterRemoveComponent)
+{
+  // Create entity
+  Entity e1 = manager.CreateEntity();
+  manager.CreateComponent<IntComponent>(e1, IntComponent(123));
+  auto e1c1 =
+    manager.CreateComponent<DoubleComponent>(e1, DoubleComponent(0.0));
+  auto e1c2 =
+    manager.CreateComponent<StringComponent>(e1, StringComponent("int"));
+
+  manager.RemoveComponent(e1, e1c1);
+  manager.RemoveComponent(e1, e1c2);
+
+  // Serialize into a message
+  msgs::SerializedStateMap stateMsg;
+  manager.State(stateMsg);
+
+  // Check message
+  {
+    auto iter = stateMsg.entities().find(e1);
+    const auto &e1Msg = iter->second;
+    auto compIter = e1Msg.components().begin();
+
+    // First component
+    const auto &e1c2Msg = compIter->second;
+    compIter++;
+    EXPECT_TRUE(e1c2Msg.remove());
+
+    // Second component
+    const auto &e1c1Msg = compIter->second;
+    compIter++;
+    EXPECT_TRUE(e1c1Msg.remove());
+
+    // Third component
+    const auto &e1c0Msg = compIter->second;
+    EXPECT_FALSE(e1c0Msg.remove());
+  }
+
+  // Check that removed components don't exist anymore
+  msgs::SerializedStateMap newStateMsg;
+  manager.State(newStateMsg);
+
+  // Check message
+  {
+    auto iter = newStateMsg.entities().find(e1);
+    const auto &e1Msg = iter->second;
+    EXPECT_EQ(1, e1Msg.components_size());
+    auto compIter = e1Msg.components().begin();
+
+    // First component
+    const auto &e1c0Msg = compIter->second;
+    EXPECT_FALSE(e1c0Msg.remove());
+  }
```

Comment:
These tests look great! One more check would be to create another instance of `EntityComponentManager` that is synced with `manager` via `SetState` before and after the components are removed. This will test that deserialization is working properly without actually publishing messages.

Reply:
It would also be great if you could add checks that cover `ClearRemovedComponents`? You'll have to add a member function `EntityCompMgrTest` that in turn calls `ClearRemovedComponents`. If you are running out of time, let me know and I can add this test.

Reply:
Thank you!
Addressed in 942c6d4.

> It would also be great if you could add checks that cover ClearRemovedComponents

Following up on https://github.com/ignitionrobotics/ign-gazebo/pull/272#discussion_r468292455, I've added a `ClearRemovedComponents` on the first two tests, so I think I don't need to add an extra test for that? Let me know.
I've also added the sync test between managers.

Reply:
That will do. Thanks!